feature : add a ProductPresenter simplified class for module developers

didpoule opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/8/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Wrap Product Presenter instantiation in a simplified class for easier using by modules devs
| Type?             | new feature
| Category?         | FO
| BC breaks?        | no
| Deprecations?     |  no
| Fixed ticket?     | Fixes #29193.
| Related PRs       | any
| How to test?      | Implement a module to get products and pass it into `ProductSimplePresenter::getInstance()->present($products)`
| Possible impacts? | no


<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

JevgenijVisockij committed:
Fixed integration test

JevgenijVisockij committed:
Apply suggestions from code review

Small fixes from code review

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

JevgenijVisockij committed:
Apply suggestions from code review

Wording domain changes

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

JevgenijVisockij committed:
Added cache clearing and removed unneeded step

JevgenijVisockij committed:
Update src/Adapter/OrderReturn/QueryHandler/GetOrderReturnForEditingHandler.php

Removed final from handler

Co-authored-by: Progi1984 <progi1984@gmail.com>

JevgenijVisockij committed:
Apply suggestions from code review

Removed some uneeded final classes, also some uneeded spaces

Co-authored-by: Progi1984 <progi1984@gmail.com>

JevgenijVisockij committed:
ADded unit tests for merchandise return

JevgenijVisockij committed:
Fixed issue with symbols < not showing as escaped values in order return customer explanation.

JevgenijVisockij committed:
Removed no longer needed javascript

JevgenijVisockij committed:
Fixed issue with wrong exception thrown when order return is not found

JevgenijVisockij committed:
Changed order state update logic to use domain services

JevgenijVisockij committed:
Fixed error codes, added error code to OrderReturnId exception. Added warning to OrderReturn create method

JevgenijVisockij committed:
Update src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/MerchandiseReturnController.php

Changed from checknig for id instead to check for isValid and is submitted when saving order return

Co-authored-by: Jonathan Lelievre <jo.lelievre@gmail.com>

JevgenijVisockij committed:
Update src/PrestaShopBundle/Resources/config/services/core/form/form_handler.yml

Using Symfony 4.4 syntax for factory

Co-authored-by: Jonathan Lelievre <jo.lelievre@gmail.com>

JevgenijVisockij committed:
Update tests/Integration/Behaviour/Features/Context/Domain/OrderReturnFeatureContext.php

Using getter rather then static function

Co-authored-by: Jonathan Lelievre <jo.lelievre@gmail.com>

JevgenijVisockij committed:
Update tests/Integration/Behaviour/Features/Scenario/OrderReturn/order_return_management.feature

Adding line break for clarity

Co-authored-by: Jonathan Lelievre <jo.lelievre@gmail.com>

JevgenijVisockij committed:
Refactored test to use order return for editing

JevgenijVisockij committed:
Renamed order return state back as it shouldn't part of this PR

JevgenijVisockij committed:
Fixed factory definition in services and removed no longer used path in forms save

JevgenijVisockij committed:
Apply suggestions from code review

Suggestions for exception comments

Co-authored-by: Progi1984 <progi1984@gmail.com>

JevgenijVisockij committed:
Apply suggestions from code review

Corrected service names

Co-authored-by: Progi1984 <progi1984@gmail.com>

JevgenijVisockij committed:
CS fixer fixes

JevgenijVisockij committed:
Added test for order return data provider

JevgenijVisockij committed:
Apply suggestions from code review

Made OrderReturn::get smaller

Co-authored-by: Progi1984 <progi1984@gmail.com>

JevgenijVisockij committed:
Fixed order return context namespace

JevgenijVisockij committed:
CS fixer fixes

JevgenijVisockij committed:
Fixed test

JevgenijVisockij committed:
Added check for duplicate order names

JevgenijVisockij committed:
Adding an if for PHP Stan to shut it

JevgenijVisockij committed:
Fixing tests

JevgenijVisockij committed:
Adding ID in front of order return state to avoid duplicate order names issues

JevgenijVisockij committed:
Using sprintf

JevgenijVisockij committed:
Adding docblock

JevgenijVisockij committed:
Fixes to yml files

JevgenijVisockij committed:
Adding missing return type

JevgenijVisockij committed:
Adding missing return type

JevgenijVisockij committed:
Update src/Adapter/OrderReturn/Repository/OrderReturnRepository.php

IMO makes sense.

Co-authored-by: Mathieu Ferment <mathieu.ferment@prestashop.com>

JevgenijVisockij committed:
Added missing )

matks committed:
Apply suggestions from code review

Co-authored-by: Progi1984 <progi1984@gmail.com>

Commit:
Fix showing price of customized products in Order Confirmation page

JevgenijVisockij committed:
Migrating base for merchandise return view page

# Conflicts:
#	src/PrestaShopBundle/Resources/config/routing/admin/sell/customer_service/merchandise_return.yml
#	src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

JevgenijVisockij committed:
Moved order return test after order

# Conflicts:
#	tests/Integration/Behaviour/behat.yml

# Conflicts:
#	tests/Integration/Behaviour/behat.yml

JevgenijVisockij committed:
Changed order return update to use customer and order repositories instead of using legacy objects outright.

Renamed EditableOrderReturn and EditableOrderReturnState into OrderReturnForEditing and OrderReturnStateForEditing

# Conflicts:
#	src/PrestaShopBundle/Resources/config/services/adapter/customer.yml
#	templates/bundles/PrestaShopBundle/Admin/Configure/ShopParameters/OrderReturnStates/edit.html.twig

JevgenijVisockij committed:
Update src/PrestaShopBundle/Resources/config/routing/admin/sell/customer_service/merchandise_return.yml

Using full notation to avoid compatability problems

Co-authored-by: Jonathan Lelievre <jo.lelievre@gmail.com>
# Conflicts:
#	src/PrestaShopBundle/Resources/config/routing/admin/sell/customer_service/merchandise_return.yml

JevgenijVisockij committed:
Removed some uneeded things like sprintf or DataProvider

# Conflicts:
#	src/PrestaShopBundle/Resources/config/routing/admin/sell/customer_service/merchandise_return.yml

JevgenijVisockij committed:
Renamed order return state back as it shouldn't part of this PR

# Conflicts:
#	templates/bundles/PrestaShopBundle/Admin/Configure/ShopParameters/OrderReturnStates/edit.html.twig

JevgenijVisockij committed:
Moved order return and order return state creation to a different context

# Conflicts:
#	tests/Integration/Behaviour/behat.yml

JevgenijVisockij committed:
CS Fixer fixes

# Conflicts:
#	src/Adapter/Customer/Repository/CustomerRepository.php

JevgenijVisockij committed:
Fixed copyright

# Conflicts:
#	src/Adapter/Customer/Repository/CustomerRepository.php

JevgenijVisockij committed:
Apply suggestions from code review

Minor fixes: return types and docs

Co-authored-by: Progi1984 <progi1984@gmail.com>
# Conflicts:
#	src/Adapter/Customer/Repository/CustomerRepository.php

Commit:
remove final class from attachment uploader


remove final class from attachment uploader

Commit:
Show errors when installing a module fails

JevgenijVisockij committed:
Removing uneeded form types

FabienPapet commented:
I don't know about how useful this feature can be or is wanted. But IMHO adding such a new feature will need to use the new architecture, use proper dependency injection, etc. 

Adding "new legacy" classes will increase the work about migrating them into the new architecture. Also, there will be some work on your side when you'll want to upgrade PrestaShop. I think it should be rewrite to use the new PS coding architecture.

JevgenijVisockij committed:
Update src/Adapter/OrderReturn/Repository/OrderReturnRepository.php

Adding extra ( just to make condition more clear

Co-authored-by: Amoifr <31698966+Amoifr@users.noreply.github.com>

JevgenijVisockij committed:
Update src/Core/Form/IdentifiableObject/DataProvider/OrderReturnFormDataProvider.php

Adding return type

Co-authored-by: Amoifr <31698966+Amoifr@users.noreply.github.com>

JevgenijVisockij committed:
Fixed $product variable

JevgenijVisockij committed:
Changed to use routerinterface

tom-combet committed:
Define empty data for mobile phone field

JevgenijVisockij committed:
Refactored repositories to work properly added integration tests for them.

JevgenijVisockij committed:
Fixed order return saving not working

JevgenijVisockij committed:
CS fixer fixes

margud committed:
fixed issue where module transaltions would take missing transaltions from default lang and not leave them empty

Commit:
Do not render empty category blocks in module manager by default

Commit:
Remove wording

JevgenijVisockij committed:
Changed validator to orderReturnValidator

JevgenijVisockij committed:
Adding missing doc block

JevgenijVisockij committed:
Update src/Adapter/Form/ChoiceProvider/OrderReturnStateChoiceProvider.php

Adding variable for clearer reading

Co-authored-by: Julius Žukauskas <31609858+zuk3975@users.noreply.github.com>

JevgenijVisockij committed:
Adding missing doc block

mflasquin committed:
Avoid possibility to create product with combinations when combinations feature is disabled

Hlavtox committed:
Allow themes to disable loading of core scripts

JevgenijVisockij committed:
Chganged repositores to use existing getObjectModel function

JevgenijVisockij committed:
Apply suggestions from code review

Some minor code review changes

Co-authored-by: Julius Žukauskas <31609858+zuk3975@users.noreply.github.com>

JevgenijVisockij committed:
Changed datetime as immutable

JevgenijVisockij committed:
Fixed test to use immutable

JevgenijVisockij committed:
Removing uneeded imports

JevgenijVisockij committed:
Removing unneeded DateTime

JevgenijVisockij committed:
Changing back not using getObjectModel function

hugofintecture committed:
Fix potential warning on getPosition method of Module class

ks129 committed:
Disable access to the customer groups page when this feature is disabled

ks129 committed:
Fix CS style

JevgenijVisockij committed:
Fixes from code review

JevgenijVisockij committed:
Update src/Core/Form/IdentifiableObject/DataProvider/OrderReturnFormDataProvider.php

Changed information whole block to be translated with variables

Co-authored-by: Jonathan Lelievre <jo.lelievre@gmail.com>

leemyongpakvn committed:
add modern and active Sortable library

as an alternative for jqueryuiSortable and jquerySortable

JevgenijVisockij committed:
Removed uneeded placeholders

JevgenijVisockij committed:
Adding change so allow html takes effect

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Type/LinkPreviewType.php

Add parent method call

Co-authored-by: Jonathan Lelievre <jo.lelievre@gmail.com>

JevgenijVisockij committed:
Enabling legacy order id

JevgenijVisockij committed:
Adding interfaces for repositories

JevgenijVisockij committed:
Added interface for state repository changes handlers to use new repositories

JevgenijVisockij committed:
Removing uneeded line

JevgenijVisockij committed:
Uncommenting legacy parameter

JevgenijVisockij committed:
Fixing cs fixer and test

JevgenijVisockij committed:
CS fixer fix

JevgenijVisockij committed:
Moving validate to state handler

JevgenijVisockij committed:
CS fixer fixes

Commit:
Remove 1 million cap limit search (ID product)

JevgenijVisockij committed:
Removed interfaces

JevgenijVisockij committed:
Updated the form to use preview form types

# Conflicts:
#	src/PrestaShopBundle/Form/Admin/Type/TextPreviewType.php
#	src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml
#	templates/bundles/PrestaShopBundle/Admin/TwigTemplateForm/prestashop_ui_kit_base.html.twig

# Conflicts:
#	src/PrestaShopBundle/Form/Admin/Type/TextPreviewType.php

# Conflicts:
#	src/PrestaShopBundle/Form/Admin/Type/TextPreviewType.php

JevgenijVisockij committed:
Refactored repositories to work properly added integration tests for them.

# Conflicts:
#	src/PrestaShopBundle/Form/Admin/Type/TextPreviewType.php
#	src/PrestaShopBundle/Resources/config/services/adapter/order_return_state.yml

# Conflicts:
#	src/PrestaShopBundle/Form/Admin/Type/TextPreviewType.php

# Conflicts:
#	src/PrestaShopBundle/Form/Admin/Type/TextPreviewType.php

JevgenijVisockij committed:
Properly adding missing repositories and twig changes.

# Conflicts:
#	admin-dev/themes/new-theme/scss/components/_form.scss

JevgenijVisockij committed:
Yasml and css fixes

# Conflicts:
#	admin-dev/themes/new-theme/scss/components/_form.scss

JevgenijVisockij committed:
Removing uneeded types

# Conflicts:
#	src/PrestaShopBundle/Form/Admin/Type/TextPreviewType.php

JevgenijVisockij committed:
Moved validation back to repository

JevgenijVisockij committed:
Cs fixer fix

JevgenijVisockij committed:
EmployeeController.php

# Conflicts:
#	tests-legacy/PrestaShopBundle/Utils/DatabaseCreator.php
#	tests/Integration/Behaviour/behat.yml

# Conflicts:
#	tests/Integration/Behaviour/behat.yml

# Conflicts:
#	tests/Integration/Behaviour/behat.yml

# Conflicts:
#	tests/Integration/Behaviour/behat.yml

JevgenijVisockij committed:
CS fixer and stand fixes

JevgenijVisockij committed:
Fixed and improved behat test a bit

# Conflicts:
#	tests/Integration/Behaviour/behat.yml

JevgenijVisockij committed:
Fixed tests for merchandise return

mflasquin committed:
WIP: Display list of authorized apps

mkaleta-waynet committed:
increase LoadServicesFromModulesPass priority

mkaleta-waynet committed:
include autoloader files from modules before compiling legacy environment containers

JevgenijVisockij committed:
Changed OrderReturnFormDataHandler.php to throw NotImplementedException.php

JevgenijVisockij committed:
Fixing translate interface

mkaleta-waynet committed:
move the LoadServicesFromModulesPass priority to a constant and add a documentation comment

Progi1984 committed:
GA : YAML Lint (YamlLint Check) : Node.js 12 actions are deprecated

Worst45 committed:
Squash commits to remove references of unchanged composer.*

JevgenijVisockij committed:
Fixed proper interface

mflasquin committed:
Rename routes

mflasquin committed:
Add style on api accesses states column

nesrineabdmouleh committed:
Fix review

mflasquin committed:
Add help link

mflasquin committed:
Applying review suggestions

mflasquin committed:
Add legacy controller

mflasquin committed:
Load default language if employee id lang does not exist

Commit:
Add sponsor company to PR description

nesrineabdmouleh committed:
Add new test duplicate product

nesrineabdmouleh committed:
Add some fixes

mflasquin committed:
Handle review to use LangageRepository instead of ObjectModel directly

mflasquin committed:
Apply some suggestions after review

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/trafficAndSeo/seoAndUrls/add`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/trafficAndSeo/seoAndUrls`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/stores`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/stores/add`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/search`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/search/add`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/search/tags/`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/search/tags/add`)

nesrineabdmouleh committed:
Fix review

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/productSettings`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/orderSettings`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/orderSettings/statuses`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/orderSettings/statuses/add`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/orderSettings/statuses/returnStatus/add`)

nesrineabdmouleh committed:
Fix consult order list test

nesrineabdmouleh committed:
Fix double steps identifiers

nesrineabdmouleh committed:
Add test scenario in comment

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/product`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/general`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/general/maintenance`)

mflasquin committed:
Don't use generic type for the column api accesses - Add some types on functions - fix route name

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/customerSettings/options`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/customerSettings/index`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/customerSettings/titles/index`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/customerSettings/titles/add`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/customerSettings/groups/index`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/customerSettings/groups/add`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/contact/index`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shopParameters/contact/add`)

Progi1984 committed:
Functional Tests : Fixed ESLint

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/shipping/preferences/index`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/quickAccess/index`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/quickAccess/add`)

nesrineabdmouleh committed:
Fix the test 'ReorderFromOrderList'

nesrineabdmouleh committed:
Add new test download invoice

nesrineabdmouleh committed:
Add some fixes

nesrineabdmouleh committed:
Add small fix

nesrineabdmouleh committed:
Fix review

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/orders/index`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/orders/add`)

zuk3975 committed:
replace and remove easily replacable/unusable methods from combinationRepository

mflasquin committed:
Add missing strict types

nesrineabdmouleh committed:
Add new test download invoice

nesrineabdmouleh committed:
Fix reorder from order detail page test

zuk3975 committed:
replace multishop repositores in supplierUpdater

zuk3975 committed:
replace last usages, remove CombinationRepository and rename MultishopRepo to CombinationRepository

FabienPapet committed:
Fix GetResponseEvent deprecations

matks committed:
Use packagist to download prestashop/autoload

Progi1984 committed:
Functional Tests : Fixed Nightly

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/orders/view/viewOrderBasePage`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/orders/view/tabListBlock`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/orders/view/productsBlock`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/orders/view/paymentBlock`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/orders/view/messagesBlock`)

Progi1984 committed:
Functional Tests : Migration TS (`@pages/BO/orders/view/customerBlock`)

Progi1984 committed:
Functional Tests : Fixed ESLint errors

nesrineabdmouleh committed:
Fix conflicting files

nesrineabdmouleh committed:
Fix filter modules by category test

nesrineabdmouleh committed:
Fix const name

dingedi committed:
Fix access for read only permissions

Commit:
Move "how to test" higher up and touch up wording

Progi1984 committed:
Functional Tests : Fixed Nightly

kpodemski committed:
remove not needed `#featuresDetachables`

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/FO/order`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/FO/shoppingCart`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/FO/account`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/shopParameters/b2b`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/orders/shoppingCarts`)

nesrineabdmouleh committed:
Test FO

zuk3975 committed:
fix preview input

nesrineabdmouleh committed:
Delete workflow

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/orderStatus`)

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/orderReturnStatus`)

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/orderMessage`)

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/language`)

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/invoice`)

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/imageType`)

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/group`)

Progi1984 committed:
Functional Tests : Migration TS (Fixed feedbacks from @boubkerbribri)

Progi1984 committed:
Functional Tests : Migration TS (Fixed Nightly)

jolelievre committed:
Turn service definitions into FQCN style for services in adapter/product.yml

jolelievre committed:
Turn service definitions into FQCN style for services in adapter/product/_combination.yml

kpodemski commented:
@FabienPapet 

This would be impossible to do for 8.1. Can we agree on an interim solution so that module developers can benefit from this change?

jolelievre committed:
Turn service definitions into FQCN style for services in adapter/product/_customization.yml

jolelievre committed:
Turn service definitions into FQCN style for services in adapter/product/_feature_value.yml

jolelievre committed:
Turn service definitions into FQCN style for services in adapter/product/_image.yml

jolelievre committed:
Turn service definitions into FQCN style for services in adapter/product/_pack.yml

jolelievre committed:
Turn service definitions into FQCN style for services in adapter/product/_seo.yml

jolelievre committed:
Turn service definitions into FQCN style for services in adapter/product/_specific_prices.yml

jolelievre committed:
Turn service definitions into FQCN style for services in adapter/product/_stock.yml

Progi1984 committed:
Functional Tests : Migration TS (Fixed commonTests)

Progi1984 committed:
Functional Tests : Migration TS (Fixed feedbacks)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/international/ecoTax`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/international/currency`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/customerService/merchandiseReturns`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/customers/customer`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/customers/address`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/catalog/monitoring`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/catalog/product`)

Commit:
Transform feature request template from GitHub issues into "idea" in Discussions

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/catalog/category`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/catalog/cartRule`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/advancedParameters/setPermissions`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/advancedParameters/newFeatures`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/advancedParameters/importFile`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/advancedParameters/cache`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/advancedParameters/employee`)

Progi1984 committed:
Functional Tests : Migration TS (`@commonTests/BO/advancedParameters/smtp`)

PululuK commented on src/Adapter/Presenter/Product/ProductSimplePresenter.php:
```diff
@@ -0,0 +1,138 @@
+        }
+
+        return self::$instance;
+    }
```
```suggestion
    public static function getInstance(Context $context = null): self
    {
        if (empty(self::$instance)) {
            self::$instance = new self($context);
        }

        return self::$instance;
    }
```

PululuK commented on src/Adapter/Presenter/Product/ProductSimplePresenter.php:
```diff
@@ -0,0 +1,138 @@
+        }
+
+        return self::$presentedProducts[$product->id];
+    }
```
```suggestion
    public function present(Product $product): ?ProductLazyArray
    {
        if (!Validate::isLoadedObject($product)) {
            return null;
        }

        if (empty(self::$presentedProducts[(int)$product->id])) {
            self::$presentedProducts[$product->id] = $this->presenter->present(
                $this->presentationSettings,
                $this->assembler->assembleProduct(['id_product' => (int)$product->id]),
                $this->context->language
            );
        }

        return self::$presentedProducts[$product->id];
    }
```

PululuK commented on src/Adapter/Presenter/Product/ProductSimplePresenter.php:
```diff
@@ -0,0 +1,138 @@
+     * @param Product[] $products
+     * @return ProductLazyArray[]|array
+     */
+    public function presentProducts(array $products)
```
```suggestion
    public function presentProducts(array $products): array
```

PululuK commented on src/Adapter/Presenter/Product/ProductSimplePresenter.php:
```diff
@@ -0,0 +1,138 @@
+     * @param mixed $product
+     * @return bool
+     */
+    public function isProduct($product)
```
```suggestion
    public function isProduct($product): bool
```

Commit:
Tone down the niceness

Co-authored-by: Tom Zajac <SharakPL@users.noreply.github.com>

PululuK reviewed (commented):
Nice job @didpoule !

A small feedbacks :)

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/file`)

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/employee`)

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/country`)

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/cmsPage`)

Progi1984 committed:
Functional Tests : Migration TS (`@data/faker/carrier`)

jolelievre committed:
Turn service definitions into FQCN style for services in adapter/product/_virtual_product.yml

jolelievre committed:
Turn service definitions into FQCN style for services in core/product.yml

jolelievre committed:
Turn service definitions into FQCN style for services in adapter/product/attribute.yml and attribute_group.yml

jolelievre committed:
Turn service definitions into FQCN style for services in adapter/feature.yml

jolelievre committed:
Adapt all definitions in product sub services to rely as much as possible on FQCN only

Progi1984 committed:
Functional Tests : Migration TS (Fixed Nightly)

jolelievre committed:
Adapt all definitions in product services to rely as much as possible on FQCN only

jbritoeg committed:
cs-fixer : method return types
